src/System.Private.CoreLib/shared/System/Resources/FastResourceComparer.cs

```diff
@@ -21,7 +22,7 @@
 
 namespace System.Resources
 {
-    internal sealed class FastResourceComparer : IComparer, IEqualityComparer, IComparer<string>, IEqualityComparer<string>
+    internal sealed class FastResourceComparer : IComparer, IEqualityComparer, IComparer<string?>, IEqualityComparer<string>
```

Comment:
Why is it an `IComparer<string?>` but an `IEqualityComparer<string>`?

Reply:
`IEqualityComparer`'s `GetHashCode` will throw on null

Reply:
Yes, but it's Equals supports nulls, right?

Reply:
so if you combine both then T cannot be null, not sure how to express that

Reply:
> so if you combine both then T cannot be null, not sure how to express that

That's one way of looking at it.  Another is that it's possible to have a single API which sometimes allows nulls and other times throws on nulls, based on the situation, and for such APIs they need to be nullable.  This is logically just another case of that.

Reply:
nullable with TODO sounds good to me